### PR TITLE
Cqm patient fixtures & script

### PR DESCRIPTION
**Pull request by hackrm**
Updated script that exports frontend patient fixtures in CQM format. New CQM formatted patient fixtures created through the rake task calls in the 'spec/javascripts/fixtures/json/README' file. Patient fixtures were not created for the measures due to issues with the version numbers, the measure no longer being available and a few other reasons: CMS134, CMS136, CMS107, CMS460, CMS529, CMS54321 & CMS890. The fixtures will replace the currently used patient fixtures in a future PR. 

Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1885
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced. N/A
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Test fixtures updated and documented as necessary ( see [internal wiki](https://gitlab.mitre.org/bonnie/internal-documentation/wikis/testing#test-fixtures) ): N/A
- [x] Code coverage has not gone down and all code touched or added is covered. 
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: 
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here: 
- [x] Automated regression test(s) pass: N/A

**Reviewer 1:**

Name: Zach Lister
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: spec/javascripts/fixtures/json/cqm_records/special_measures/CMS334v1/patients.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[
+
+]
```
Comment (hackrm) [inline spec/javascripts/fixtures/json/cqm_records/special_measures/CMS334v1/patients.json:3]: **The original fixture for this measure's patients is also blank.
File path to the original fixture: https://github.com/projecttacoma/bonnie/blob/master/spec/javascripts/fixtures/json/records/special_measures/CMS334v1/patients.json

File: spec/javascripts/fixtures/json/cqm_records/special_measures/cms720/patients.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[
+
+]
```
Comment (hackrm) [inline spec/javascripts/fixtures/json/cqm_records/special_measures/cms720/patients.json:3]: **The original fixture for this measure's patients is also blank.
File path to the original fixture: https://github.com/projecttacoma/bonnie/blob/master/spec/javascripts/fixtures/json/records/special_measures/CMS720/patients.json

File: spec/javascripts/fixtures/json/README [hunk 3]
```diff
@@ -25,15 +25,13 @@
 
 
 # CQL-based measures
-CMS136 created with:
+!! This measure does not exist on this account - CMS136 created with:
 bundle exec rake bonnie:fixtures:generate_frontend_cql_fixtures[cms,CQL/CMS136,epecqmncqa@gmail.com,CMS741v0,"BehaviorHealth<=30DaysAftrADHDMed","NUMERPop2Pass"]
 
 CMS146 created with:
-bundle exec rake bonnie:fixtures:generate_frontend_cql_fixtures[cms,CQL/CMS146,epecqmncqa@gmail.com,CMS146v5,"PharynStrepATest<=3dB4Enc","NUMERPass"]
-# TODO: when cql-testing-overhaul is merged, re-export CMS146 and remove CQL/CMS146v6.
```
Comment (hackrm) [inline spec/javascripts/fixtures/json/README:33]: This TODO comment has been addressed and the CMS 146 fixtures have been re-exported within this PR.

File: lib/tasks/cql_testing.rake [hunk 4]
```diff
@@ -125,17 +125,6 @@ namespace :bonnie do
       end
     end
 
-    ###
-    # Creates and writes a fixture file.
-    #
-    # file_path: path of file to be written
-    # fixture_json: json to be written
-    def create_fixture_file(file_path, fixture_json)
```
Comment (hackrm) [inline lib/tasks/cql_testing.rake:133]: This function has been moved to the fixture_exporter.rb file: https://github.com/projecttacoma/bonnie/blob/471ce8eeca1ece82d8094048e7dffbff1a514be8/lib/util/fixture_exporter.rb

File: lib/tasks/cql_testing.rake [hunk 5]
```diff
@@ -19,7 +19,7 @@ namespace :bonnie do
     task :generate_frontend_cql_fixtures, [:cms_hqmf, :path, :user_email, :measure_id, :patient_first_name, :patient_last_name] => [:environment] do |t, args|
       fixtures_path = File.join('spec', 'javascripts', 'fixtures', 'json')
       measure_file_path = File.join(fixtures_path, 'measure_data', args[:path])
-      record_file_path = File.join(fixtures_path, 'records', args[:path])
+      record_file_path = File.join(fixtures_path, 'cqm_records', args[:path])
```
Comment (hackrm) [inline lib/tasks/cql_testing.rake:22]: CQM patient fixtures will now be stored in the directory: spec/javascripts/fixtures/json/cqm_records

File: lib/util/fixture_exporter.rb [hunk 6]
```diff
@@ -14,39 +14,39 @@ def export_measure_and_any_components(path)
     measure = as_transformed_hash(@measure)
     measure_name = @measure.cms_id + '.json'
     measure_file = File.join(path, measure_name)
-    create_fixture_file(measure_file, JSON.pretty_generate(measure))
```
Comment (hackrm) [inline lib/util/fixture_exporter.rb:17]: This JSON.pretty_generate() call is now called within the create_fixture_file() function instead.

File: lib/util/fixture_exporter.rb [hunk 7]
```diff
@@ -139,12 +146,19 @@ def objectid_to_oids(input)
...
   def initialize(user, measure: nil, records: nil)
     super(user, measure: measure, records: records)
   end
 
+  def export_records_as_array(path)
+    @records = @records.map { |record| CQMConverter.to_cqm(record) }
+    super(path)
+  end
+
+  # Alias must be placed after the export_records_as_attay method so super method won't be called first
```
Comment (zlister) [inline lib/util/fixture_exporter.rb]: typo, should be export_records_as_array* not export_records_as_attay

**Review by zlister — APPROVED** [review]

**Review by jbradl11 — APPROVED** [review]
